### PR TITLE
Fixed session Persistence including adding /logout endpoint

### LINKED ISSUE #46: Adjust authentication to ensure proper persistence across sessions
Current sessions may not be properly persisted across multiple requests, which may lead to issues once we being implementing more authenticated endpoints.

### PR DESCRIPTION
**Pull request by AminZeina**
### Additions
- User sessions are now properly stored and persisted across requests in the backend by saving the authenticated context in a spring `securityContextRepository` (see the documenation [here](https://docs.spring.io/spring-security/reference/servlet/authentication/session-management.html#store-authentication-manually) for more details)
  - This means that once a user is logged it, any subsequent requests with automatically be authenticated as that user
A `/logout` endpoint is now available to log out the current user, implemented using Spring's `SecurityContextLogoutHandler` defined in our `/config/SecurityConfig.java` (see the documentation [here](https://docs.spring.io/spring-security/reference/servlet/authentication/logout.html) for more details)
  - `/logout` is a POST endpoint (with an empty body) and will return `204 - No Content` on successful logout
  - This endpoint needed to be included to close #46 properly as without it we would be unable to clear sessions on the server side
  - Note that the endpoint is automatically created by the `SecurityContextLogoutHandler` which is why it's not explicitly defined anywhere a Controller
- The logout button in the MenuBar now makes a call to `/logout` to logout the user on the server-side

Since this is my first shot at playing with the frontend, let me know if how I modified the MenuBar to add the api call makes sense and if there's anything I should change/a better way of doing this!

This closes #46. Sorry for the double PR for this branch, I forgot to add that last commit 🤦‍♂️

### CODE REVIEW

File: frontend/src/MenuBar.jsx [hunk 1]
```diff
@@ -1,17 +1,33 @@
...
+    const toast = useRef(null);
     const [loginVisible, setLoginVisible] = useState(false);
     const [user, setUser] = useContext(UserContext);
 
-    const handleLogout = () => {
+    const handleLogout = async () => {
+
+        const logoutStatus = await logoutUser();
         setUser(null);
+        console.log("point a")
```
Comment (yehandesilva) [inline frontend/src/MenuBar.jsx]: Remove this console.log()

File: frontend/src/MenuBar.jsx [hunk 2]
```diff
@@ -1,17 +1,33 @@
...
+        if (logoutStatus.success) {
+            setUser(null);
+            console.log("point b")
```
Comment (yehandesilva) [inline frontend/src/MenuBar.jsx]: Remove this console.log()

**Review by yehandesilva — CHANGES_REQUESTED** [review]
Looks good and the new logout route makes sense. You've properly integrated it with the frontend and the toast component to show error messages is properly configured. Just need to remove some console.log() statements and its good to merge.

**Comment by AminZeina** [comment]
Thanks for the feedback, good to know it all looks good on the frontend side of things! For the console logs, I did actually remove them in commit [a1a7be3](https://github.com/yehandesilva/monkeyPoll/pull/52/commits/a1a7be3afb0e2880a23b5f9d4da28d509294bbcd) so it seems like it's showing you outdated code (might be because I also created #51 without [a1a7be3](https://github.com/yehandesilva/monkeyPoll/pull/52/commits/a1a7be3afb0e2880a23b5f9d4da28d509294bbcd) which messed things up). 

But anyways could you double check that with the most recent changes that the console logs are removed?

**Review by pathumd — APPROVED** [review]
Changes look good Amin! Manually tested creating a new Account, logging in, and logging out and can confirm no console.log messages were shown.

**Review by yehandesilva — APPROVED** [review]
Don't see the loggin messages anymore. Good to merge!

**Pull request merged**